File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	var total int
	for i := 1; i < limit; i++ {
		for _, divisor := range divisors {
			if divisor != 0 && i%divisor == 0 {
				total += i
				break
			}
		}
	}
	return total
}

File: solutions/c/sum-of-multiples/1/sum_of_multiples.c
#include "sum_of_multiples.h"

unsigned int sum(const unsigned int *factors, const size_t number_of_factors, const unsigned int limit) {
    if (number_of_factors == 0) { 
        return 0; 
    }

    unsigned int total = 0;

    for (unsigned int n = 1; n < limit; n++) {
        for (size_t i = 0; i < number_of_factors; i++) {
            if (factors[i] != 0 && n % factors[i] == 0) {
                total += n;
                break;
            }
        }
    }
    return total;
}
